src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Tokenization.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Intellisense;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Intellisense
+{
+    /// <summary>
+    /// An entity with methods to compute ordered or unordered enumeration enumeration of token text spans in a expression rule with their start and end indices and token type.
+    /// </summary>
+    internal static class Tokenization
+    {
+        /// <summary>
+        /// Returns an ordered enumeration of token text spans in a expression rule with their start and end indices and token type. Orders token by default comparer.
+        /// </summary>
+        /// <param name="expression">Expression which would be tokenized.</param>
+        /// <param name="binding">Binding which would be used to tokenize the operation and determine the type of each token.</param>
+        /// <param name="comments">Colllection of comment tokens extracted from the given expression.</param>
+        /// <returns>Ordered collection of tokens.</returns>
+        internal static ICollection<ITokenTextSpan> Tokenize(string expression, TexlBinding binding, IEnumerable<CommentToken> comments = null)
```

Comment:
This overload would be mostly used by old formula bar once these changes are consumed